File: resumos/cursos/cs50/all-challenges/problems/sentimental/readability.py
from cs50 import get_string

text = get_string("Text: ")


def calculateGrade(text):
    length = len(text)
    countLetters = 0
    countWords = 0
    countSentences = 0

    if text[0].isalpha:
        countWords += 1

    for i in range(length):
        unicodeLetter = ord(text[i])

        if text[i].isalpha():
            countLetters += 1
        elif unicodeLetter == 32:
            if text[i+1].isalpha() or text[i+2].isalpha():
                countWords += 1

        if unicodeLetter == 46 or unicodeLetter == 33 or unicodeLetter == 63:
            countSentences += 1

    L = countLetters / countWords * 100
    S = countSentences / countWords * 100
    grade = round(0.0588 * L - 0.296 * S - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


calculateGrade(text)

File: resumos/cursos/cs50/all-challenges/problems/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int calculateGrade(string text);

int main(void)
{
    string text = get_string("Please type a text: ");
    int resultGrade = calculateGrade(text);
}

int calculateGrade(string text)
{
    int length = strlen(text);
    int countLetters = 0;
    int countWords = 0;
    int countSentences = 0;

    if (isalpha(text[0]))
    {
        countWords++;
    }

    // Pass for every letter in a input
    for (int i = 0; i < length; i++)
    {
        // If is a letter count based on is alphanumeric
        if (isalpha(text[i]))
        {
            countLetters++;
        }

        // Finded a space? Check if the next character is a letter, then count a word
        else if (text[i] == 32)
        {

            // Avoid "", but I think it can be better...
            if (isalpha(text[i + 1]) || isalpha(text[i + 2]))
            {
                countWords++;
            }
        }

        // Check if character is a !,? or ., then end a sentence.
        else if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            countSentences++;
        }

    }

    // Calculate Coleman-Liau index
    float L = (float)countLetters / (float)countWords * 100;
    float S = (float)countSentences / (float)countWords * 100;
    float grade = round(0.0588 * L - 0.296 * S - 15.8);

    if ((int) grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if ((int) grade >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }

    printf("Grade %i\n", (int) grade);

    return 0;
}